### ISSUE TITLE
jsgrid custom field and jquery chosen

### ISSUE THREAD

**Issue by bjornivarpb** [tabalinas/jsgrid#775]
Hi i'm using jsgrid and it's really great. I'm just having trouble with changing the current select filter to jquery chosen.  I created a custom field to implement it but it doesn't work. Chosen works fine on other parts of my web app (like inside a jquery dialog), i just can't make it work inside jsgrid.  Is this the correct approach? Thank you in advance.

Here is my code:
1)This is the jsGrid field that i want to convert to jquery chosen

```js
{
 name: "ID", title: "Rep", type: "select", items: repList, valueField: "ID", textField: "Nome", align: "left",
  filterTemplate: function () {
     var $chosenSel = jsGrid.fields.select.prototype.filterTemplate.apply(this);
     //$chosenSel.addClass("text ui-widget-content ui-corner-all"); 
     $chosenSel.addClass("chzn-select");
    return $chosenSel; 
    }
}
```

2)

```js
 $(".chzn-select").chosen({
   'no_results_text': 'Nessun record...',
  });
```

3) links:

```html
 <script src="https://cdnjs.cloudflare.com/ajax/libs/chosen/1.4.2/chosen.jquery.js"></script>
 <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/chosen/1.4.2/chosen.css"/>  
```

**Comment by tabalinas** [comment]
Did you check out this issue https://github.com/tabalinas/jsgrid/issues/512?

I think defining a custom field would be a good option (while it's not the simplest one). 
Probably your approach would work better if you need only filtering. Please, provide more details on what doesn't work exactly with your approach.

**Comment by bjornivarpb** [comment]
Hi, 

Thank you for your response. Yes i've seen that issue but i don't see a sample on how he changed the select control to jquery chosen. Indeed I thought it would just be as simple as using .addClass like the code below:

{
 name: "ID", title: "User Type", type: "select", items: userTypes, valueField: "ID", textField: "Nome", align: "left",
 filterTemplate: function () {
var $multisel = jsGrid.fields.select.prototype.filterTemplate.apply(this);
$multisel.addClass("text ui-widget-content ui-corner-all");
return $multisel;
}

I tried to do it this way, but this does not work.
{
 name: "ID", title: "User Type", type: "select", items: userTypes, valueField: "ID", textField: "Nome", align: "left",
filterTemplate: function () {
var $chosenSel = jsGrid.fields.select.prototype.filterTemplate.apply(this);
$chosenSel.addClass("chzn-select");
return $chosenSel; 
    }
}

I just need to apply chosen to filter controls only, because i don't currently use the inline add/edit. I use jquery modal pop up for that. I hope I made my concern clear :) Thanks in advance for any help. 

![jsgrid](https://cloud.githubusercontent.com/assets/28621630/26544791/7377cb4a-4463-11e7-82f0-b41ffaaffe61.JPG)
